full-service/src/db/txo.rs

```diff
@@ -519,14 +527,30 @@ impl TxoModel for Txo {
             // Find the account associated with this Txo.
             // Note: We should only be calling update_to_pending on inputs, which we had to
             // own to spend.
-            let account = Account::get_by_txo_id(&txo_id.to_string(), conn)?;
+            let accounts = Account::get_by_txo_id(&txo_id.to_string(), conn)?;
 
             // Update the status to pending.
-            diesel::update(
-                account_txo_statuses.find((&account.account_id_hex, &txo_id.to_string())),
-            )
-            .set(crate::db::schema::account_txo_statuses::txo_status.eq(TXO_PENDING.to_string()))
-            .execute(conn)?;
+            if accounts.len() > 2 {
```

Comment:
Handle the case where a Txo "belongs" to two accounts in the wallet (it was minted by one account and received by another).